src/pages/index.js

```diff
@@ -2,18 +2,102 @@ import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import PopuuWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
 import {
-  initialCards,
+  avatarButton,
+  avatarImage,
   config,
   editProfileModalDescriptionInput,
   editProfileModalTitleInput,
   cardAddBtn,
   profileEditBtn,
   selectors,
+  apiInfo,
 } from "../utils/utils.js";
 import "../pages/index.css";
+const { apiToken, apiUrl } = apiInfo;
+let userId = null;
+let section = null;
+
+const api = new Api({
+  baseUrl: apiUrl,
+  headers: {
+    authorization: apiToken,
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo(
+  selectors.userNameSelector,
+  selectors.userJobSelector,
+  selectors.userAvatarSelector,
+);
+
+const handleDeleteClick = (card) => {
+  confirmationModal.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: you should also show the loding process by changing the confirm button text